controlplane/kubeadm/internal/control_plane.go

```diff
@@ -385,3 +389,79 @@ func (c *ControlPlane) InjectTestManagementCluster(managementCluster ManagementC
 	c.managementCluster = managementCluster
 	c.workloadCluster = nil
 }
+
+// StatusToLogKeyAndValues returns following key/value pairs describing the overall status of the control plane:
+// - machines is the list of KCP machines; each machine might have additional notes surfacing
+//   - if the machine has been created in the current reconcile (new)
+//   - if machines node is not yet (node ref not set)
+//   - if the machine has bee marked for remediation (health check failed)
+//   - if there are unhealthy control plane component on the machine
+//   - if the machine has a deletion timestamp/has been deleted in the current reconcile (deleting)
+//   - if the machine is not up to date with the KCP spec (not up to date)
+//
+// - etcdMembers list as reported by etcd.
+func (c *ControlPlane) StatusToLogKeyAndValues(newMachine, deletedMachine *clusterv1.Machine) []any {
+	controlPlaneMachineHealthConditions := []clusterv1.ConditionType{
+		controlplanev1.MachineAPIServerPodHealthyCondition,
+		controlplanev1.MachineControllerManagerPodHealthyCondition,
+		controlplanev1.MachineSchedulerPodHealthyCondition,
+	}
+	if c.IsEtcdManaged() {
+		controlPlaneMachineHealthConditions = append(controlPlaneMachineHealthConditions,
+			controlplanev1.MachineEtcdPodHealthyCondition,
+			controlplanev1.MachineEtcdMemberHealthyCondition,
+		)
+	}
+
+	machines := []string{}
+	for _, m := range c.Machines {
+		notes := []string{}
+
+		if m.Status.NodeRef == nil {
+			notes = append(notes, "node ref not set")
+		}
+
+		if c.MachinesToBeRemediatedByKCP().Has(m) {
+			notes = append(notes, "health check failed")
+		}
+
+		for _, condition := range controlPlaneMachineHealthConditions {
+			if conditions.IsUnknown(m, condition) {
+				notes = append(notes, strings.Replace(string(condition), "Healthy", " health unknown", -1))
+			}
+			if conditions.IsFalse(m, condition) {
+				notes = append(notes, strings.Replace(string(condition), "Healthy", " not healthy", -1))
+			}
+		}
+
+		if !c.UpToDateMachines().Has(m) {
+			notes = append(notes, "not up to date")
+		}
+
+		if !m.DeletionTimestamp.IsZero() || (deletedMachine != nil && m.Name == deletedMachine.Name) {
+			notes = append(notes, "deleting")
+		}
+
+		name := m.Name
+		if len(notes) > 0 {
+			name = fmt.Sprintf("%s (%s)", name, strings.Join(notes, ", "))
+		}
+		machines = append(machines, name)
+	}
+
+	if newMachine != nil {
+		machines = append(machines, fmt.Sprintf("%s (new)", newMachine.Name))
```

Comment:
```suggestion
		machines = append(machines, fmt.Sprintf("%s (created)", newMachine.Name))
```
WDYT about created (or just created)? Just fits better into creationTimestamp/deletionTimestamp <=> create/delete

Reply:
Also added Just deleted